python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1231,6 +1231,11 @@ def _construct_run_with_snapshots(
             else None
         )
 
+        if job_snapshot:
+            from ..op_concurrency_limits_counter import compute_concurrency_tags_for_snapshot
+
+            tags = {**tags, **compute_concurrency_tags_for_snapshot(job_snapshot)}
```

Comment:
one thing we might consider doing is to put this as a field on the runs, rather than polluting the tags table with a bunch of json garbage.